AssociativeTable.cs

```diff
@@ -54,49 +48,40 @@ public void Put(TKey tempKey, TValue tempValue)
     /// Returns all values of Tuple elements inside Data array with passed key value.
     /// </summary>
     /// <param name="tempKey">The key to search for.</param>
-    public TValue Get(TKey tempKey)
+    public override TValue Get(TKey tempKey)
     {
         for (int i = 0; i < Count; i++)
         {
-            if (EqualityComparer<TKey>.Default.Equals(Data[i].Item1, tempKey))
+            if (EqualityComparer<TKey>.Default.Equals(Data[i].Key, tempKey))
             {
-                return Data[i].Item2;
+                return Data[i].Value;
             }  
         }
         throw new Exception("Key not found");
     }
-    
-    /// <summary>
-    /// Removes the specified key and its associated value from the hash map.
-    /// </summary>
-    /// <param name="tempValue">The Value to remove - method will loop through whole array.</param>
-    public void RemoveByValue(TValue tempValue)
+
+    public override bool ContainsKey(TKey tempKey)
     {
-        int tempIndex = -1;
         for (int i = 0; i < Count; i++)
         {
-            if (EqualityComparer<TValue>.Default.Equals(Data[i].Item2, tempValue))
+            if (EqualityComparer<TKey>.Default.Equals(Data[i].Key, tempKey))
             {
-                tempIndex = i;
-                for (int j = tempIndex; j < Data.Length; j++)
-                {
-                    Data[i] = Data[i + 1];
-                    Count--;
-                }
+                return true;
             }
         }
+        return false;
     }
     
     /// <summary>
     /// Removes the specified key and its associated value from the hash map.
     /// </summary>
     /// <param name="tempKey">The key to remove - method will loop through whole array..</param>
-    public void RemoveByKey(TKey tempKey)
+    public override void Remove(TKey tempKey)
```

Comment:
lets change the name of parameters `tempKey` to just `key`, remember to update the docstring

Reply:
maybe instead of two for loops which are O(n^2) let's create an instance's attribute for holding index of the first empty node and let's use one for loop that seeks the node, removes it if it finds it and changes the value of the attribute if the index of the node was lower than the assigned to it. we should also modify the put method if we decide to change this mechanism